include/secp256k1.h

```diff
@@ -127,6 +127,10 @@ typedef int (*secp256k1_nonce_function)(
 #  define SECP256K1_INLINE inline
 # endif
 
+#ifndef SECP256K1_BUILD
+# define SECP256K1_NO_BUILD
+#endif
```

Comment:
nit: I think this would be a good place to spend one or two sentences to explain the purpose of these macros. (My personal experience: I think I had two ask a few times before I understood the entire concept.)

Reply:
There is a comment below on the nullness macro. But I can add one to the guard about the guard specifically.